velox/functions/sparksql/tests/ArrayGetTest.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayGetTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  void arrayGet(
+      const ArrayVectorPtr& arrayVector,
+      const int32_t index,
+      const std::optional<T> expectValue) {
+    auto res = evaluate(
+        "get(c0, c1)",
+        makeRowVector({arrayVector, makeConstant(index, arrayVector->size())}));
+    assertEqualVectors(makeNullableFlatVector<T>({expectValue}), res);
+  }
+};
+
+TEST_F(ArrayGetTest, basic) {
+  auto arrayVector = makeNullableArrayVector<int32_t>({{1, 2, std::nullopt}});
+  arrayGet<int32_t>(arrayVector, -1, std::nullopt);
+  arrayGet<int32_t>(arrayVector, 0, 1);
+  arrayGet<int32_t>(arrayVector, 1, 2);
```

Comment:
Please test this case and add it to document

```
spark-sql (default)> SELECT get(array(1, 2, 3),null);
NULL

```

Reply:
added.